src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -731,6 +731,7 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+                    requestUrlPath = PathNormalizer.NormalizeToNFC(requestUrlPath);
```

Comment:
`NormalizeToNFC` is not about `.` and `/`. It's about this: http://www.macchiato.com/unicode/nfc-faq

In theory we could implement the equivalents of `string.IsNormalized` and `string.Normalize` for Kestrel, using memory blocks. Unfortunately, we don't have time for it now :disappointed: